src/widgets/components/clustering_steps.py

```diff
@@ -0,0 +1,216 @@
+        self.description_label.setWordWrap(True)
+
+        self.description_group_box = QGroupBox()
+        self.description_group_box.setFixedWidth(250)
+        self.description_group_box.setTitle("Description")
+        self.description_group_box_layout = QVBoxLayout(self.description_group_box)
+
+        self.scroll_box = QGroupBox()
+        self.scroll_box_layout = QFormLayout(self.scroll_box)
+        self.scroll = QScrollArea()
+        self.scroll.setHorizontalScrollBarPolicy(Qt.ScrollBarAlwaysOff)
+        self.scroll.setVerticalScrollBarPolicy(Qt.ScrollBarAsNeeded)
+        self.scroll.setSizePolicy(QSizePolicy.Expanding, QSizePolicy.Expanding)
+        self.scroll.setWidget(self.scroll_box)
+        self.scroll.setWidgetResizable(True)
+        self.scroll.setMinimumHeight(26)
+        self.description_group_box_layout.addWidget(self.scroll)
+
+        self.scroll_box_layout.addWidget(self.description_label)
+
+        self.left_column_layout.addWidget(self.description_group_box, 1)
+
+        self.left_column_layout.setSpacing(35)
+
+        self.layout.addLayout(self.left_column_layout)
+        self.layout.addWidget(self.visualization_box)
+
+    def click_listener(self, button_type: str):
+        match button_type:
+            case "parameters_changed":
+                self.ox = self.parameters_widget.ox
+                self.oy = self.parameters_widget.oy
+                self.samples = self.parameters_widget.samples
+                self.parameters_changed.emit()
+            case "prev":
+                num = self.left_box.value()
+                self.change_step(-1 * num)
+            case "next":
+                num = self.right_box.value()
+                self.change_step(num)
+            case "restart":
+                self.animation = None
+                self.change_step(-1 * self.current_step)
+                self.change_enabled_buttons(True)
+                self.interval_box.setEnabled(True)
+                self.run_button.setEnabled(True)
+            case "run":
+                self.is_running = not self.is_running
+                self.change_enabled_buttons(False)
+                self.interval_box.setEnabled(False)
+                self.restart_button.setEnabled(not self.is_running)
+                if self.is_running:
+                    if self.animation is None:
+                        self.animation = self.get_func_animation()
+                        self.canvas.draw()
+                    else:
+                        self.animation.resume()
+                    self.run_button.setText("Stop animation")
+                else:
+                    self.animation.pause()
+                    self.run_button.setText("Start animation")
+
+    def change_enabled_buttons(self, value):
+        self.ox_box.setEnabled(value)
+        self.oy_box.setEnabled(value)
+        self.sample_button.setEnabled(value)
+        self.sample_box.setEnabled(value)
```

Comment:
these attributes are now in `self.parameters_widget`, this method should be moved as well